### DISCUSSION TITLE
Langserve Templates

### DISCUSSION THREAD

**Discussion by anusha2310-netizen** [langchain-ai/langserve#321] [Q&A]
I have a few tools like web search and knowledge base search and ticket classification as a conversational agent where depends upon the query llm decides which tool to pickup, in order to leverage this to langserve using templates, what are the templates I have to use and is there any option where I can use multiple templates as a conversational agent, please help

**Comment by eyurtsev** [comment]
Hi @anusha2310-netizen,

Take a look at the templates here: https://github.com/langchain-ai/langchain/tree/master/templates

You're probably looking for an agent template. The openai agent is probably the best starting point for using an agent.

**Comment by anusha2310-netizen** [comment]
yeah I am using openai agent but can we use multiple templates as a conversational agent?

**Comment by eyurtsev** [comment]
My understanding is that it'll be useful to have a template that shows how to use an OpenAI assistant with a few tools in a conversational style -- and that's missing right now?

---

You can use multiple templates at the same time. Add the templates to your app and just make sure that you register them under different paths when doing `add_routes(..., path= "[use a unique path per chain]" )` that will expose relevant endpoints associated with each template.

Templates are reference implementations for specific patterns etc. Different templates may implement the same architecture using different architectural pieces (e.g., different vector database), or may show completely different architectures but for achieving the same underlying task.

---

Roughly it sounds like what you need is:

1. Make sure that the tools for web search / knowledge base and ticket classification are working correctly as tools.
2. Create an open ai agent with access to the tools above.
3. Hook it up via langserve

You can follow docs here to create the agent: https://python.langchain.com/docs/modules/agents/agent_types/openai_functions_agent#using-openaifunctionsagent

With access to relevant tools, and then you should be able to expose it via add_routes in langserve.
